clients/ui/frontend/src/app/pages/modelRegistry/screens/ModelVersions/ModelDetailsCard.tsx

```diff
@@ -0,0 +1,119 @@
+import { ModelRegistryContext } from '~/app/context/ModelRegistryContext';
+
+type ModelDetailsCardProps = {
+  registeredModel: RegisteredModel;
+  refresh: () => void;
+  isArchiveModel?: boolean;
+};
+
+const ModelDetailsCard: React.FC<ModelDetailsCardProps> = ({
+  registeredModel: rm,
+  refresh,
+  isArchiveModel,
+}) => {
+  const { apiState } = React.useContext(ModelRegistryContext);
+
+  return (
+    <Card>
+      <CardTitle>Model details</CardTitle>
+      <CardBody>
+        <Stack hasGutter>
+          <StackItem>
+            <DescriptionList>
+              <ModelEditableLabelsDescriptionListGroup
+                isArchiveModel={isArchiveModel}
+                rm={rm}
+                refresh={refresh}
+              />
+              <EditableTextDescriptionListGroup
+                editableVariant="TextArea"
+                baseTestId="model-description"
+                title="Description"
+                isArchive={isArchiveModel}
+                contentWhenEmpty="No description"
+                value={rm.description || ''}
+                saveEditedValue={(value) =>
+                  apiState.api
+                    .patchRegisteredModel(
+                      {},
+                      {
+                        description: value,
+                      },
+                      rm.id,
+                    )
+                    .then(refresh)
+                }
+              />
+            </DescriptionList>
+          </StackItem>
+          <StackItem>
+            <DescriptionList columnModifier={{ default: '1Col', md: '2Col' }}>
+              <DashboardDescriptionListGroup
+                title="Owner"
+                popover="The owner is the user who registered the model."
+              >
+                <Content component="p" data-testid="registered-model-owner">
+                  {rm.owner || '-'}
+                </Content>
+              </DashboardDescriptionListGroup>
+              <DashboardDescriptionListGroup title="Model ID">
+                <ClipboardCopy
+                  hoverTip="Copy"
+                  clickTip="Copied"
+                  variant="inline-compact"
+                  data-testid="registered-model-id-clipboard-copy"
+                >
+                  {rm.id}
+                </ClipboardCopy>
+              </DashboardDescriptionListGroup>
+              <DashboardDescriptionListGroup
+                isEmpty={!rm.lastUpdateTimeSinceEpoch}
+                contentWhenEmpty="Unknown"
+                title="Last modified at"
+              >
+                <ModelTimestamp timeSinceEpoch={rm.lastUpdateTimeSinceEpoch} />
+              </DashboardDescriptionListGroup>
+              <DashboardDescriptionListGroup
+                isEmpty={!rm.createTimeSinceEpoch}
+                contentWhenEmpty="Unknown"
+                title="Created at"
```

Comment:
This will conflict with #1368 @Taj010 FYI you need to rebase if this goes in first else @DaoDaoNoCode has to rebase if the other PR goes in first. Same with "Last modified at"